pkgs/active-modules/default.nix

```
@@ -0,0 +1,85 @@
+# active-modules is a hydrated version of `nix eval .#modules --json` which contains for each module
+# * name
+# * description
+# * tag
+# * version
+# * all historical tags
+# meant to be used by the module registry UI
+
+# to see for yourself, do:
+# nix build .#active-modules
+# cat result | jq
+
+# or
+# nix eval .#active-modules.meta.info --json | jq
+  
+{ self, pkgs }:
+with pkgs.lib;
+let
+  active-modules = self.modules;
+  get-version = (tag:
+    let module = active-modules.${module-id};
+        tag-parts = strings.splitString "-" tag;
+        version-str = elemAt tag-parts 0;
+        version = toInt (substring 1 (stringLength version-str) version-str);
+    in version
+  );
+  all-modules = (builtins.fromJSON (builtins.readFile ../../modules.json));
+  all-modules-list = (attrsets.mapAttrsToList (name: value: { registry-id = name; commit = value.commit; path = value.path; }) all-modules);
+  active-modules-registry = 
+  foldr (
+    entry: registry:
+      let
+        parts = strings.splitString ":" entry.registry-id;
+        module-id = elemAt parts 0;
+        tag = elemAt parts 1;
+        version = get-version tag;
+        module-info = (builtins.fromJSON (builtins.unsafeDiscardStringContext active-modules.${module-id}.text));
+        new-entry = {
+          inherit (entry) commit path;
+          inherit version;
+          inherit tag;
+          inherit (module-info) name description;
+        };
+      in
+      if ! builtins.hasAttr module-id active-modules then
+        registry
+      else
+        if builtins.hasAttr module-id registry then
+          let prev-entry = registry.${module-id};
+          in
+          if prev-entry.version > new-entry.version then
+            registry // {
+              ${module-id} = prev-entry // {
+                tags = prev-entry.tags ++ [tag];
+              };
+            }
+          else
+            registry // {
+              ${module-id} = new-entry // {
+                tags = prev-entry.tags ++ [tag];
+              };
+            }
+        else
+          registry // {
+            ${module-id} = new-entry // {
+              tags = [tag];
+            };
+          }
+    )
+    {}
+    all-modules-list;
+in
+pkgs.stdenv.mkDerivation {
+  pname = "active-modules";
+  version = "1";
+  src = ./.;
+  installPhase = ''
+  echo '${builtins.toJSON active-modules-registry}' > $out
+  '';
+  meta = {
```

```suggestion
  passthru = {
```

meta is supposed to be for metadata about the package. passthru should do what you want.